openaz-xacml/src/main/java/org/apache/openaz/xacml/std/jaxp/JaxpRequest.java

```diff
@@ -114,19 +123,13 @@ public static JaxpRequest load(File fileXmlRequest) throws ParserConfigurationEx
             logger.error("No Document returned parsing \"" + fileXmlRequest.getAbsolutePath() + "\"");
             return null;
         }
-
-        NodeList nodeListRoot = document.getChildNodes();
-        if (nodeListRoot == null || nodeListRoot.getLength() == 0) {
-            logger.warn("No child elements of the XML document");
-            return null;
+        
+        Node nodeRoot = document.getDocumentElement();
+        
+        if(context == null) {
+            init();
```

Comment:
What i was saying, you already have a state defined `context` of a class `JaxpRequest` , what i don't like you are modifying state of this class, better create  separate Singelton class   and `JaxpRequest` will get instance of a context from that singelton class (global access) would be easier for testing aslo. Although, singleton is a bad practice but you need some sort of life cyle management outside request level.

I don't like you are adding init method in the request level which should be outside request management. I don't like another aspect of adding synchronization in the code.